Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+        
+    # tetrahedron unit vectors
+    x1=np.array([1,1,1])
+    x2=np.array([-1,1,-1])
+    x3=np.array([1,-1,-1])
+    x4=np.array([-1,-1,1])
+
+    substituentMoment= LA.norm((subSizes[0]*x1)+(subSizes[1]*x2)+(subSizes[2]*x3)+(subSizes[3]*x4))
+    return substituentMoment
+
+def generateChiralDescriptors(mol):
+    desc={}
+    dists = Chem.GetDistanceMatrix(mol)
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    subDict, sn_dict, maxShell_dict = determineAtomSubstituents(idxChiral, mol, dists, False)
+    sizes = calcSizeSubstituents(mol, subDict, sn_dict, maxShell_dict)
+    paths = dists[idxChiral]
+    # set some basic descriptors
+    desc['numAtoms'] = mol.GetNumAtoms()
+    desc['numBonds'] = mol.GetNumBonds()
+    desc['numRotBonds'] = AllChem.CalcNumRotatableBonds(mol)
+    desc['ringChiralCenter'] = int(mol.GetAtomWithIdx(idxChiral).IsInRing())
+    # determine the max path length in the molecule and the mean pairwise distance of all atom pairs
+    desc['meanDist'] = np.sum(dists)/((desc['numAtoms']-1)*(desc['numAtoms']))
+    desc['maxDist'] = int(np.max(dists))
+    # determine the max path length from the chiral center and the mean pairwise distance of 
+    # all atom pairs from the chiral center
+    desc['meanDistFromCC'] = np.sum(paths)/(desc['numAtoms']-1)
+    desc['maxDistfromCC'] = int(np.max(paths))
+    # determine the number of neighbors per shell/distance level
+    nlevels=Counter(paths.astype(int))
+    # consider the levels until a path lenght of 10
+    for i in range(1,11):
+        desc['nLevel'+str(i)]=nlevels[i]
+    # determine the number of nitrogen and oxygen atoms in a certain level around the chiral center
+    for i in range(1,4):
+        desc['phLevel'+str(i)]=len([n for n,j in enumerate(paths) if j==i and mol.GetAtomWithIdx(n).GetAtomicNum() in [7,8]])
+    # determine the number of arometic atoms in a certain level around the chiral center
+    for i in range(1,4):
+        desc['arLevel'+str(i)]=len([n for n,j in enumerate(paths) if j==i and mol.GetAtomWithIdx(n).GetIsAromatic()])
+    # set the size descriptors for each substituent, sort them from smallest to largest
+    for n,v in enumerate(sorted(sizes.items(), key=lambda x: x[1].size)):
+        desc['s'+str(n+1)+'_size'] = v[1].size
+        desc['s'+str(n+1)+'_relSize'] = v[1].relSize
+        desc['s'+str(n+1)+'_phSize'] = v[1].numNO
+        desc['s'+str(n+1)+'_phRelSize'] = v[1].relNumNO
+        desc['s'+str(n+1)+'_phRelSize_2'] = v[1].relNumNO_2
+        desc['s'+str(n+1)+'_pathLength'] = v[1].pathLength
+        desc['s'+str(n+1)+'_relPathLength'] = v[1].relPathLength
+        desc['s'+str(n+1)+'_relPathLength_2'] = v[1].relPathLength_2
+        desc['s'+str(n+1)+'_numSharedNeighbors']=v[1].sharedNeighbors
+        desc['s'+str(n+1)+'_numRotBonds']=v[1].numRotBonds
+        desc['s'+str(n+1)+'_numAroBonds']=v[1].numAroBonds
```

Comment:
`sizes` is a dictionary and you sort it by the values. In the loop, you only use `v[1]` of each dictionary item, which is the value. So instead of sorting items, sort the values:
```
for n, v in enumerate(sorted(sizes.values(), key=lambda x: x.size)):
  desc['s'+str(n+1)+'_size'] = v.size
  ...
```
In addition, you want to create variables that start with 1, e.g. s1_size. The `enumerate` function takes a second, optional argument which gives the start of the counter
```
for n, v in enumerate(sorted(sizes.values(), key=lambda x: x.size), 1):
  sn = 's' + str(n)
  desc[sn+'_size'] = v.size
  ...
```